julia.html.markdown

```diff
@@ -145,171 +145,394 @@ SomeOtherVar123! = 6 # => 6
 filled_dict["one"] # => 1
 
 # Get all keys
-keys(filled_dict)
-# => KeyIterator{Dict{ASCIIString,Int64}}(["three"=>3,"one"=>1,"two"=>2])
+keys(filled_dict) # => Base.KeyIterator for a Dict{String,Int64} with 3 entries.
+                  #    Keys:
+                  #     "two"
+                  #     "one"
+                  #     "three"
 # Note - dictionary keys are not sorted or in the order you inserted them.
 
 # Get all values
-values(filled_dict)
-# => ValueIterator{Dict{ASCIIString,Int64}}(["three"=>3,"one"=>1,"two"=>2])
+values(filled_dict) # => Base.ValueIterator for a Dict{String,Int64} with
+                    #    3 entries. Values:
+                    #     2
+                    #     1
+                    #     3
 # Note - Same as above regarding key ordering.
 
 # Check for existence of keys in a dictionary with in, haskey
 in(("one" => 1), filled_dict) # => true
-in(("two" => 3), filled_dict) # => false
+("two" => 3) in filled_dict # => false
 haskey(filled_dict, "one") # => true
 haskey(filled_dict, 1) # => false
 
 # Trying to look up a non-existent key will raise an error
 try
-    filled_dict["four"] # => ERROR: key not found: four in getindex at dict.jl:489
+    filled_dict["four"] # => KeyError: key "four" not found
 catch e
     println(e)
 end
 
 # Use the get method to avoid that error by providing a default value
 # get(dictionary,key,default_value)
-get(filled_dict,"one",4) # => 1
-get(filled_dict,"four",4) # => 4
+get(filled_dict, "one", 4) # => 1
+get(filled_dict, "four", 4) # => 4
 
 # Use Sets to represent collections of unordered, unique values
-empty_set = Set() # => Set{Any}()
+empty_set = Set() # => Set{Any}[]
 # Initialize a set with values
-filled_set = Set([1,2,2,3,4]) # => Set{Int64}(1,2,3,4)
+filled_set = Set([1,2,2,3,4]) # => Set([4, 2, 3, 1])
 
 # Add more values to a set
-push!(filled_set,5) # => Set{Int64}(5,4,2,3,1)
+push!(filled_set, 5) # => Set{Int64}([4,2,3,5,1])
+filled_set # => Set{Int64}([4,2,3,5,1])
 
 # Check if the values are in the set
-in(2, filled_set) # => true
-in(10, filled_set) # => false
+2 in filled_set # => true
+10 in filled_set # => false
 
 # There are functions for set intersection, union, and difference.
-other_set = Set([3, 4, 5, 6]) # => Set{Int64}(6,4,5,3)
-intersect(filled_set, other_set) # => Set{Int64}(3,4,5)
-union(filled_set, other_set) # => Set{Int64}(1,2,3,4,5,6)
-setdiff(Set([1,2,3,4]),Set([2,3,5])) # => Set{Int64}(1,4)
+other_set = Set([3, 4, 5, 6]) # => Set{Int64}([4,3,5,6])
+intersect(filled_set, other_set) # => Set{Int64}([4,3,5])
+filled_set ∩ other_set # => Set{Int64}([4,3,5]) # The same with ∩ (\cap)
+union(filled_set, other_set) # => Set{Int64}([4,2,3,5,6,1])
+filled_set ∪ other_set # => Set{Int64}([4,2,3,5,6,1]) # The same ∪ (\cup)
+setdiff(Set([1,2,3,4]), Set([2,3,5])) # => Set{Int64}([4,1])
+
+# Set operations work on arrays, too
+[1,2,3,4] ∩ [2,3,4] # => 3-element Array{Int64,1}: [2,3,4]
+
+
+
+
+###########
+## Strings
+###########
```

Comment:
This ends up being explained after strings have been used (i.e. as keys). What's preventing them from being explained earlier? Seems more important than collections.

Reply:
The 5e60a12 fixed that, but it did not get recognized automatically.